pallets/claim/src/tests.rs

```diff
@@ -0,0 +1,584 @@
+    })
+}
+
+#[test]
+fn cannot_start_new_aidrop_if_one_already_in_progress() {
+    test().execute_with(|| {
+        assert_ok!(Claim::begin_airdrop(
+            Origin::Signed(MANAGER_USER).into(),
+            None
+        ));
+        assert_evt(Event::AirdropStarted { airdrop_id: 0 }, "New airdrop");
+        assert_err!(
+            Claim::begin_airdrop(
+                Origin::Signed(MANAGER_USER).into(),
+                Some(GENESIS_BENEFICIARIES_MAP.clone())
+            ),
+            Error::<Test>::AlreadyStarted
+        );
+        assert_not_evt(Event::AirdropStarted { airdrop_id: 1 }, "No new airdrop");
+    })
+}
+
+#[test]
+fn claim() {
+    test_with_configs(
+        WithGenesisBeneficiaries::Yes,
+        GenesisClaimBalance::Sufficient,
+    )
+    .execute_with(|| {
+        assert_ok!(Claim::claim(Origin::Signed(USER_1).into(), None));
+        assert_evt(
+            Event::Claimed {
+                beneficiary: USER_1,
+                amount: USER_1_AMOUNT,
+                payment_id: (),
+            },
+            "Successfull claim",
+        );
+        assert_eq!(Claim::pot(), SUFFICIENT_GENESIS_BALANCE - USER_1_AMOUNT);
+        assert_eq!(
+            TotalClaimable::<Test>::get(),
+            SUFFICIENT_GENESIS_BALANCE - USER_1_AMOUNT
+        );
+        assert_eq!(Balances::free_balance(USER_1), USER_1_AMOUNT);
+        assert!(Beneficiaries::<Test>::get(USER_1).is_none());
+    });
+}
+
+#[test]
+fn double_claim_is_err() {
+    test_with_configs(
+        WithGenesisBeneficiaries::Yes,
+        GenesisClaimBalance::Sufficient,
+    )
+    .execute_with(|| {
+        assert_ok!(Claim::claim(Origin::Signed(USER_1).into(), None));
+        assert_err!(
+            Claim::claim(Origin::Signed(USER_1).into(), None),
+            Error::<Test>::NotEligible
+        )
+    });
+}
+
+#[test]
+fn claim_with_opt_dest() {
+    test_with_configs(
+        WithGenesisBeneficiaries::Yes,
+        GenesisClaimBalance::Sufficient,
+    )
+    .execute_with(|| {
+        assert_ok!(Claim::claim(Origin::Signed(USER_1).into(), Some(USER_2)));
+        assert_evt(
+            Event::Claimed {
+                beneficiary: USER_2,
+                amount: USER_1_AMOUNT,
+                payment_id: (),
+            },
+            "Successfull claim for a different dest",
+        );
+        assert_eq!(Claim::pot(), SUFFICIENT_GENESIS_BALANCE - USER_1_AMOUNT);
+        assert_eq!(
+            TotalClaimable::<Test>::get(),
+            SUFFICIENT_GENESIS_BALANCE - USER_1_AMOUNT
+        );
+        assert_eq!(Balances::free_balance(USER_1), 0);
+        assert_eq!(Balances::free_balance(USER_2), USER_1_AMOUNT);
+        assert!(Beneficiaries::<Test>::get(USER_1).is_none());
+    });
+}
+
+#[test]
+fn claim_wrong_beneficiary() {
+    test_with_configs(
+        WithGenesisBeneficiaries::Yes,
+        GenesisClaimBalance::Sufficient,
+    )
+    .execute_with(|| {
+        assert_err!(
```

Comment:
`assert_nop!` checks that you have an error **and** the storage not change: that's enough to check everything.